openquake/hazardlib/geo/surface/base.py

```diff
@@ -225,6 +239,79 @@ def get_joyner_boore_distance(self, mesh):
         """
         return self.get_mesh().get_joyner_boore_distance(mesh)
 
+    def get_ry0_distance(self, mesh, option=1):
+        """
+        :param mesh:
+            :class:`~openquake.hazardlib.geo.mesh.Mesh` of points to calculate
+            Ry0-distance to.
+        :parameter option:
```

Comment:
The one that Norm recommended to use is the default. The other one is my old implementation. I left this there just for experimentation. It's accessible only via the hazardlib.